runtime/include/localeModels/flat/chpl-locale-model.h

```diff
@@ -66,4 +66,17 @@ c_sublocid_t chpl_rt_sublocFromLocaleID(chpl_localeID_t loc) {
   return c_sublocid_any;
 }
 
+//
+// These functions are exported from the locale model for use by
+// the tasking layer to convert between a full sublocale and an
+// execution sublocale.
+//
+extern
+c_sublocid_t chpl_localeModel_sublocToExecutionSubloc(
```

Comment:
Got it.  Yes, if the "requested subloc" for a task needs to include the full information, then because they're responsible for storing them in task-private memory, the runtime tasking layers have to be able to assemble and disassemble them even though they don't need to know what the assembled form looks like.  Now that I get that, I'm okay with this at least for now.  I have in mind an inkling of what might be a better way to do module-oriented task-private data, not just for this but other things like serial state, but I'm not sure I can get that done for the release and definitely this PR shouldn't be held up for it.  I'll make a design issue for that.